SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/SpatialPingComponent.cpp

```diff
@@ -181,6 +182,48 @@ void USpatialPingComponent::OnRep_ReplicatedPingID()
 	}
 }
 
+FSpatialPingAverageData USpatialPingComponent::GetAverageData() const
+{
+	FSpatialPingAverageData Data = {};
+
+	if (LastPingMeasurements.Num() > 0)
+	{
+		double Total = 0.0;
+		for (float Ping : LastPingMeasurements)
+		{
+			Total += Ping;
+		}
+		Data.LastMeasurementsWindowAvg = Total / LastPingMeasurements.Num();
+		Data.LastMeasurementsWindowMin = FMath::Min(LastPingMeasurements);
+		Data.LastMeasurementsWindowMax = FMath::Max(LastPingMeasurements);
+	}
+	Data.WindowSize = LastPingMeasurements.Num();
+
+	Data.TotalAvg = TotalPing / TotalNum;
```

Comment:
TotalNum can be 0, converted to float but someone might turn float exceptions on

Reply:
Good catch, will update!